core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterStopFailedHandler.java

```diff
@@ -0,0 +1,69 @@
+package com.sequenceiq.cloudbreak.reactor.handler.cluster;
+
+import static com.sequenceiq.cloudbreak.api.endpoint.v4.common.Status.STOP_FAILED;
+import static com.sequenceiq.cloudbreak.event.ResourceEvent.CLUSTER_STOP_FAILED;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.common.event.Selectable;
+import com.sequenceiq.cloudbreak.conclusion.ConclusionChecker;
+import com.sequenceiq.cloudbreak.conclusion.ConclusionCheckerFactory;
+import com.sequenceiq.cloudbreak.conclusion.ConclusionCheckerType;
+import com.sequenceiq.cloudbreak.conclusion.ConclusionResult;
+import com.sequenceiq.cloudbreak.core.flow2.cluster.stop.ClusterStopEvent;
+import com.sequenceiq.cloudbreak.core.flow2.stack.CloudbreakFlowMessageService;
+import com.sequenceiq.cloudbreak.reactor.api.event.StackEvent;
+import com.sequenceiq.cloudbreak.reactor.api.event.cluster.ClusterStopFailedRequest;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.ExceptionCatcherEventHandler;
+import com.sequenceiq.flow.reactor.api.handler.HandlerEvent;
+
+import reactor.bus.Event;
+
+@Component
+public class ClusterStopFailedHandler extends ExceptionCatcherEventHandler<ClusterStopFailedRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClusterStopFailedHandler.class);
+
+    @Inject
+    private ConclusionCheckerFactory conclusionCheckerFactory;
+
+    @Inject
+    private CloudbreakFlowMessageService flowMessageService;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    @Override
+    public String selector() {
+        return EventSelectorUtil.selector(ClusterStopFailedRequest.class);
+    }
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<ClusterStopFailedRequest> event) {
+        return new StackEvent(ClusterStopEvent.FINALIZED_EVENT.event(), resourceId);
+    }
+
+    @Override
+    protected Selectable doAccept(HandlerEvent<ClusterStopFailedRequest> event) {
+        ClusterStopFailedRequest request = event.getData();
+        LOGGER.info("Handle ClusterStopFailedRequest, stackId: {}", request.getResourceId());
+        try {
+            ConclusionChecker conclusionChecker = conclusionCheckerFactory.getConclusionChecker(ConclusionCheckerType.DEFAULT);
+            ConclusionResult conclusionResult = conclusionChecker.doCheck(request.getResourceId());
+            if (entitlementService.conclusionCheckerSendUserEventEnabled(ThreadBasedUserCrnProvider.getAccountId()) && conclusionResult.isFailureFound()) {
```

Comment:
wouldn't be better to move the event sending to the end of conclusionChecker.doCheck method?